native-audio/app/src/main/java/com/example/nativeaudio/NativeAudio.java

```diff
@@ -33,9 +37,12 @@
 import android.widget.Spinner;
 import android.widget.Toast;
 
-public class NativeAudio extends Activity {
+public class NativeAudio extends Activity
+        implements ActivityCompat.OnRequestPermissionsResultCallback {
 
     //static final String TAG = "NativeAudio";
+    private static final int NATIVE_AUDIO_REQUEST = 0;
+    private Boolean permissionGranted = true;
```

Comment:
you are looking at the old code from here.  Latest code already addressed this and you have reviewed above. No more review necessary for any code from here and down below  -- they are older version of the same code. It is kind of github thing -- unsmart thing :-)